core/src/jsMain/kotlin/external/JsIterator.kt

```diff
@@ -0,0 +1,25 @@
+package com.juul.kable.external
+
+internal external interface JsIterator<T> {
+    fun next(): JsIteratorResult<T>
+}
+
+internal external interface JsIteratorResult<T> {
```

Comment:
"internal external" Haha. I get it, but funny.

Reply:
Question on this, though: "external" means, "the implementation is defined outside of kotlin," right?

If that's the case, how're we indicating to kotlin where that definition is?

Reply:
> If that's the case, how're we indicating to kotlin where that definition is?

We actually don't have to. Since the Kotlin is transpiled to JavaScript, it will be running in the actually JavaScript environment (in the browser in our case). So we're simply promising the compiler that this function signature exists, and for us, it will exist when everything is transpiled. So the transpiled code will call this `interface` and as long as it actually matches the signature that does in fact exist, then all is good.

Reply:
FYI this protocol is part of the ECMAScript 2015 which is why we can define it as `external` and fully expect it to exist in the target runtime (for conforming browsers).

https://docs.w3cub.com/javascript/iteration_protocols#The_iterator_protocol